File: lib/commands/builtin/windows.ts
import type { CommandSpec } from "../types"

export const windowsCommand: CommandSpec = {
  name: "windows",
  aliases: ["wins"],
  summary: "list browser windows",
  usage: ["windows", "wins"],
  man: [
    "NAME",
    "  windows, wins - list browser windows",
    "",
    "SYNOPSIS",
    "  windows | wins",
    "",
    "OUTPUT",
    "  One line per window: optional leading * if focused, then the active",
    "  tab title only (no window id or type)."
  ],
  async execute(ctx) {
    return ctx.listWindows()
  }
}

File: wasm/bmxt-core/src/cmd/windows.rs
use crate::meta::Cmd;
use crate::model::{DispatchJson, Effect};

pub const CMD: Cmd = Cmd {
    name: "windows",
    aliases: &["wins"],
    usage_primary: "windows",
    man: &[
        "NAME",
        "  windows, wins - list browser windows",
        "",
        "SYNOPSIS",
        "  windows | wins",
        "",
        "OUTPUT",
        "  One line per window: optional leading * if focused, then the active",
        "  tab title only (no window id or type).",
    ],
};

pub fn run(_args: &[String]) -> DispatchJson {
    DispatchJson::effects(vec![Effect::ListWindows])
}
